rubrikinc/rubrik-extension-for-vcd
# B1.1.0

# Description

Release of v1.1.0

## Motivation and Context

Patch Notes - 1.1.0
### Existing Features
* On Demand Snapshots
* Assign SLA Protection
* Recover vApp
* File Recovery
* Export vApp
* Credential Management

### New Features for 1.1.0
* Dashboard Feature
    * Protected vApps
    * SLA Summary
    * Summary Graphs
* Reporting 
    * SLA Compliance - Details
    * Capacity Over Time - Details
    * Protection/Recovery Tasks - Details
* Username Encryption Added for Metadata
* Branding/White-label added
* Ability to see Events per vApp
* Event Series - In depth task details
* Provider Authentication Moved inside Modal
* Rubrik Organisation Support - Global and Org both supported

### Bug Fixes
* Fixed a bug with metadata updating/deleting
* Fixed a bug with vApps containing `+` symbols
* Fixed Issues with Tenant access to 2x vCD Cells in Rubrik with a same named vApp
* Fixed Issues with SLAs not loading - Rubrik Version (4.2 and 5.0+) detection added including SLAv2 Support
* Re-written Data Population for Protection Data-grid - async callbacks
* Polaris Managed SLAs temporarily hidden - Breaking On-Demand and SLA Assignment
* File Recovery Form Crashes after running Export (Refresh to Fix)
* Export Showing Fields to name vApp when Restoring to existing
* Export Not resetting all fields on cancel

## How Has This Been Tested?

Extensively in the SE Lab and TME Lab.

Tests carried out:
* Tenant automatically authenticates without needing settings form
* Test Features Hide/Show Depending on Metadata
* Test All Graphs Display Correctly
* Check Wizards all Re-Open after submit and Cancel on pages
* Perform Export of partial vApp to New vApp
* Authenticate as System/Provider
* Add Credentials for Tenant
* vApps All display regardless of SLA or state
* Assign an SLA to an already protected vApp
* Assign an SLA to 'Not Protected' vApp
* Assign multiple SLAs to both Unprotected and Protected vApps
* Perform vApp On-Demand Snapshot
* Perform Multiple vApp On-Demand Snapshots
* Perform Instant Recovery of vApp for Single VM - Delete All NICs
* Perform Instant Recovery of vApp for Multiple/All VMs
* Perform Instant Recovery with No Mapping
* Perform File Recovery with Guest Credentials with Overwrite
* Perform File Recovery with Guest Credentials with Alternate Location
* Test All Reports load data
* Check Username and Passwords are encrypted in Metadata
* Check Features are set in Metadata
* View Event Series for vApp
* Perform Instant Recovery with Advanced Networks
* Perform Export of partial vApp to Existing vApp
* Perform Instant Recovery with Delete all NICs
* Perform Export of full vApp to New vApp - Advanced
* Perform File Recovery with RBS with Overwrite
* Perform File Recovery with RBS with Alternate Location

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

## Checklist:

Go over all the following points, and put an `x` in all the boxes that apply. If you're unsure about any of these, don't hesitate to ask. We're here to help!
- [X] I have read the **[CONTRIBUTION](/master/CONTRIBUTING.md)** document.
- [X] I have updated the CHANGELOG file accordingly for the version that this merge modifies.
